### PR TITLE
test: replace test-secret-mount with gcloud

### PR DESCRIPTION
**Pull request by tam7t**
Attempt to simplify integration tests by replacing the test-secret-mount job with something that looks more like the `examples/mypod.yaml`

Instead of scheduling a job that reads the secret from filesystem and writes to configmap - just schedule a pod and exec into the pod to read filesystem contents. This removes 1 container to build and maintain and should make it easier to write more complicated test cases like rotation.

### CODE REVIEW

File: test/infra/prow/presubmit.sh [hunk 1]
```diff
@@ -27,11 +27,6 @@ export SECRET_STORE_VERSION=v0.0.18
 # https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md#pod-utilities
 export GCP_PROVIDER_SHA=${GCP_PROVIDER_SHA:-$PULL_PULL_SHA}
 
-# Use the SA token in environment variable to run gCloud commands if provided
-if [ -n "${GOOGLE_APPLICATION_CREDENTIALS+set}" ]; then
-    gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}
-fi
-
```
Comment (tam7t) [inline test/infra/prow/presubmit.sh:34]: workload identity is used now

File: test/e2e/e2e-test-job.yaml.tmpl [hunk 2]
```diff
@@ -17,6 +17,7 @@ metadata:
   name: $JOB_NAME
   namespace: e2e-test
 spec:
+  backoffLimit: 0
```
Comment (tam7t) [inline test/e2e/e2e-test-job.yaml.tmpl:20]: the default limit is 6 retries. I could see maybe bumping it to 1 to get 1 automatic retry on a failed integration test, but largely failures seem to be real failures (and can be retried using `/retest`) so set to 0 initially

Comment (rlingutla) [reply test/e2e/e2e-test-job.yaml.tmpl:20]: This sounds good to me.

File: test/e2e/README.md [hunk 3]
```diff
@@ -4,10 +4,6 @@ This directory contains E2E tests that can be run as a job on a cluster. See [te
 
 E2E tests rely on [Config Connector](https://cloud.google.com/config-connector/docs/overview) to setup and tear-down test GKE clusters and assume that it is available.
 
-## Test Secret Mounter
-
-E2E tests run a simple test binary (test secret mounter) that reads a secret from a predefined location on the filesystem and writes it to a k8s configmap.
```
Comment (rlingutla) [inline test/e2e/README.md:9]: Nit: 'test-secret-mounter' (to match the actual binary name)

**Review by rlingutla — APPROVED** [review]

**Pull request merged**